waiter/src/waiter/scheduler/marathon.clj

```diff
@@ -275,7 +275,9 @@
      :cpus cpus
      :healthChecks [{:protocol (-> (or health-check-proto backend-proto) hu/backend-proto->scheme str/upper-case)
                      :path health-check-url
-                     :gracePeriodSeconds grace-period-secs
+                     :gracePeriodSeconds (if (pos? grace-period-secs)
+                                           grace-period-secs
+                                           (-> instance-expiry-mins t/minutes t/in-seconds))
```

Comment:
Won't this cause the instance to immediately die right when we expect to mark it as expired? That seems to defeat the purpose of expiry. Maybe we should do 1.5 times the expiry period, or just add some fixed amount of time as a buffer (e.g., 5 minutes).

This is the scenario I'm thinking of:

The instance just expired, so we're going to seamlessly create a replacement. But now the database goes down for 30 seconds, so the app is now unhealthy. Since the grace period has passed, Marathon immediately kills the expired instance without waiting for the replacement. The behavior we want is for the expired instance to stay around until the replacement is ready in order to avoid downtime. In this case, maybe the app takes 2 minutes to start a new instance, so now we have a 2 minute downtime instead of just 30 seconds caused by the database blip.

Reply:
I just noticed that Steven's original recommendation was `instance-expiry-mins*60`. I think 2x the expiry would be fine (i.e., it "double-expired" so we're ok to kill it now). I'd be worried that if we pick some fixed interval like I suggested as an alternative (e.g., 5 minutes) that it would be too short and we'd end up in the same scenario that I described above.

Reply:
Increased to twice the `instance-expiry-mins`